src/test/java/com/github/martapietka/ppm2pbmSpring/Ppm2PbmControllerTest.java

```diff
@@ -35,14 +35,20 @@ class Ppm2PbmControllerTest {
     public void shouldReturnCorrectImage() throws Exception {
 
         String imageName = "3x3.ppm";
-        byte[] expectedBytes = Files.readAllBytes(Paths.get("src/test/resources/" + imageName));
-        String expectedContentType = "image/x-portable-graymap";
+        String expectedContentType = "image/png";
 
-        given(imageService.getImage(anyString())).willReturn(new ByteArrayResource(expectedBytes));
+        InputStream inputStream = getClass().getResourceAsStream("/" + imageName);
+        assert inputStream != null;
+        byte[] expectedBytes = inputStream.readAllBytes();
 
-        this.mockMvc.perform(get("/image/{imageName}", imageName))
+        given(imageService.getImage(any(InputStream.class))).willReturn(new ByteArrayResource(expectedBytes));
```

Comment:
Nie ma przeciwwskazań by użyć tej samej tablicy bajtów na wejściu i wyjściu z serwisu, natomiast zauważ, że nie masz pewności, co tak naprawdę w przekazanym tam streamie jest.

To ogólnie jest jak najbardziej w porządku i w wielu wypadkach można by go było tak po prostu zostawić, ale w ramach ćwiczenia zainteresuj się mechanizmem _argument captor_. Pozwala on przechwycić argumenty przekazywane do mockowanych metod i wykonać na nich asercje (np. w tym przypadku: że przekazany stream zawiera tablicę bajtów `expectedBytes`).

**Uwaga: powyższa linijka NIE WYMAGA ZMIANY**, argument captorów używa się w fazie weryfikacji, razem z asercjami, wraz z wykorzystaniem metody `verify(...)` (też z `Mockito`) która sprawdza czy metoda została wywołana i czy została wywołana ze wskazanymi argumentami.